solr/core/src/java/org/apache/solr/handler/admin/SecurityConfHandler.java

```diff
@@ -260,22 +260,25 @@ public String toString() {
 
   @Override
   public Collection<Api> getApis() {
-    if (apis == null) {
+    Collection<Api> syncedApis = apis;
+    if (syncedApis == null) {
       synchronized (this) {
-        if (apis == null) {
-          Collection<Api> apis = new ArrayList<>();
+        syncedApis = apis;
+        if (syncedApis == null) {
+          syncedApis = new ArrayList<>();
+          apis = syncedApis;
```

Comment:
Is this correct? Doesn't this line assigning syncedApis to apis expose the partially configured empty list to the rest of the world as a non-immutable collection while the rest of this method executes (which was the original complaint)? I think without this line the patch is an improvement...(the existing code has this problem too).